uktrade/data-hub-frontend
# Add Opportunities field to interaction form

## Description of change

This PR introduces the addition of the Opportunities fields into the create interactions form. Recording this information comes in the shape of 2 questions on the Interactions form, a yes/no as to whether the interaction relates to an opportunity, then a typeahead that lets you choose one opportunity it's attached to.  This is necessary so that we can link an interaction to an opportunity, and also is the precursor to work around allowing users to see interactions related to an opportunity on the opportunity's end of things. 

## Test instructions

Head to the deployment of the branch below - go to companies, click on a company and make a new interaction. You should see a field at the bottom of the main form (after the first step of the create interaction journey) which asks if this interaction is related to a large capital opportunity. Click yes, and you should see a typeahead field come up, allowing you to search for and select existing opportunities. You should be able to submit the form and be redirected to a table that shows you the details of the interaction you've just created. You should see in this table the details of the opportunity you selected, along with a working link taking you to the correct opportunity. You should be able to click on edit interaction and see your previous selections in the fields as initial values. 

## Screenshots

<img width="1165" alt="Screenshot 2021-09-09 at 16 56 37" src="https://user-images.githubusercontent.com/46787754/132721186-9b706219-9631-4a1a-a704-0e0ea256a2cd.png">
<img width="936" alt="Screenshot 2021-09-09 at 17 00 22" src="https://user-images.githubusercontent.com/46787754/132721198-4abdc97b-c3b3-405f-9a52-cf631b76383e.png">

## Checklist

[//]: # "When submitting a PR make sure the code review guidelines have been satisfied.
https://github.com/uktrade/data-hub-frontend/blob/master/docs/Code%20review%20guidelines.md"
